main_test.go

```diff
@@ -630,49 +630,6 @@ func TestNewTLSConfig(t *testing.T) {
 	}
 }
 
-func TestGetMaxResponseTime(t *testing.T) {
```

Comment:
Could you add tests verifying `config.TimeoutCfg` fields are correctly set for various input configs?

Reply:
Please, see update